docs/guides/upgrade.md

```diff
@@ -67,7 +67,7 @@ There is currently now automatic way of porting over old configuration settings
 > :warning: **IMPORTANT WARNING** :warning:
 > 
 > **Before you restart** a node, please perform due diligence when porting over the node's `rank` value.
-> 1) It's possible that the rank specified in the `config.toml` doesn't match the validators **current** rank anymore. Please, check the logs to see which rank the validator is currently at. You'll need a minimum log level of INFO for this.
```

Comment:
This is extremely important. Right now, SignCTRL doesn't save its last rank when it terminates. Here's an example:

* A starts with rank 1
* B starts with rank 2
* A becomes unavailable and terminates eventually
* B notices A missed blocks, so it promotes itself to rank 1 and starts signing

If you were start up A again with its initial rank 1, both A and B would be signing simultaneously.

Reply:
But on a upgrad you would restart all nodes, and then it wouldn't matter right?

Reply:
Correct. But, on the off chance that the network is flaky and a rank update occurs in the middle of restarting the nodes, it's possible that a node's rank is rendered obsolete and leads to double-signing. Granted, that's a really tight window for that to happen, and frankkly very unlikely, but I'd still mention it. Rather be safe than sorry.

Reply:
But if that is the case, we might need to offer a way of easily checking the current rank, and recommend stopping nodes from lowest to highest rank?

Reply:
I can imagine a `signctrl status` command that would just print out the current rank and missed blocks in a row. This would remove the need of having to look into the logs.

The order in which the nodes are upgraded doesn't matter. What matters is that a node is not started with an obsolete rank. Here's an example how this would happen:

Stop validator (rank 2) to update binary -> Rank update is triggered (so it should be rank 1) -> Start Validator (but it's rank 2, not 1)

Reply:
Okay, so I will leave it as is, and we'll improve it in the future